app/scripts/modules/core/src/presentation/modal/TabBoundary.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { useEventListener } from 'core';
+
+export const TabBoundary = ({ children }: { children: React.ReactNode }) => {
+  const dialogRef = React.createRef<HTMLDivElement>();
+
+  const keydownCallback = (event: KeyboardEvent) => {
+    const { keyCode, shiftKey, target } = event;
+    if (keyCode === 9) {
+      let targetOverride = null;
+      // tabbable selector via https://kennethbi.com/posts/5
+      const tabbableElements: HTMLElement[] = Array.from(
+        dialogRef.current.querySelectorAll(`
```

Comment:
do we need some null coalescing on `dialogRef.current` for document events that happen pre-mount? I'm not totally clear on how the lifecycles interact.

Reply:
Possibly? It doesn't hurt to add a null check. I think, since the list is computed every time the listener fires, it should always have a `current` value, but who can say.